src/pkg/packager/variables.go

```diff
@@ -101,19 +102,20 @@ func (p *Packager) setVariableMapInConfig() error {
 				return err
 			}
 
-			p.setVariableInConfig(variable.Name, val, variable.Sensitive, variable.AutoIndent)
+			p.setVariableInConfig(variable.Name, val, variable.Sensitive, variable.AutoIndent, variable.Type)
 		}
 	}
 
 	return nil
 }
 
-func (p *Packager) setVariableInConfig(name, value string, sensitive bool, autoIndent bool) {
+func (p *Packager) setVariableInConfig(name, value string, sensitive bool, autoIndent bool, varType string) {
```

Comment:
What are your thoughts on simplifying this function to just take a `types.ZarfSetVariable` as the only input. Looking at how we call this function now, be practically always just decompose a ZarfSetVariable anyways.

Reply:
I did it this way so that it would be forced to be updated everywhere that it was called (similar to a New() function) - we could implement an actual New() pattern but simply making it a ZarfSetVariable might be ripe for someone to forget to set one of its fields in one of the callers.